netket/driver/vmc.py

```diff
@@ -120,12 +115,12 @@ def _forward_and_backward(self):
         self._loss_stats, self._loss_grad = self.state.expect_and_grad(self._ham)
 
         if self.preconditioner is not None:
-            self._S = self.preconditioner.object(self.state)
+            self._S = self.preconditioner[0](self.state)
```

Comment:
what was the argument against named tuples? I think that using [0] etc might be potentially much more bug prone

Reply:
The argument is that if we have a namedtuple, you cannot do

```python
gs = nk.VMC(..., preconditioner = (QGTWhatever, jax.scipy.linalg.cg)) 
```

but you must do

```python
gs = nk.VMC(..., preconditioner = nk.optimizer.Preconditioner(solver=QGTWhatever, object=jax.scipy.linalg.cg)) 
```
or create the namedtuple yourself